File: Learning/OtherLanguages/C/C练习题/Use C/4.c
#include <stdio.h>
#include <stdlib.h>

void main()
{
    int day, month, year, sum=0, leap=0;
    printf("Please input year, month, day:\n");
    scanf("%d%d%d", &year, &month, &day);

    switch(month)
    {
        case 1  : sum = 0; break;
        case 2  : sum = 31; break;
        case 3  : sum = 59; break;
        case 4  : sum = 90; break;
        case 5  : sum = 120; break;
        case 6  : sum = 151; break;
        case 7  : sum = 181; break;
        case 8  : sum = 212; break;
        case 9  : sum = 243; break;
        case 10 : sum = 273; break;
        case 11 : sum = 304; break;
        case 12 : sum = 334; break;
        default : printf("data error!\n"); exit(1);
    }

    sum += day;
    if(0 == year%400 || (0 == year%4 && 0 != year%100))
    {
        leap = 1;
        //printf("%d is a leap year~\n", year);
    }
    if(leap && month>=3)
    {
        sum +=1;
    }
    printf("The day is the %dth day of the year!\n", sum);
}

File: Learning/OtherLanguages/C/C练习题/Use Python/4.py

y = int(input('请输入年：'))
m = int(input('请输入月：'))
d = int(input('请输入日：'))

#建立月份对应天数增加的字典   实现了类似C语言中 switch...case的功能
month_dict = {1:0, 2:31, 3:59, 4:90, 5:120, 6:151, 7:181, \
              8:212, 8:243, 10:273, 11:304, 12:334}

#判断是否为闰年
if y % 400 == 0 or (y % 4 == 0 and y % 100 != 0):  #闰年
    day = month_dict[m] + d + 1
    print('该年是闰年')
else:  #非闰年
    day = month_dict[m] + d
    print('该年不是闰年')
print('您所输入的日期为该年的第%d的天' % day)


    
